src/components/container/login/loginComponent.jsx

```diff
@@ -0,0 +1,180 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { toast } from 'react-toastify';
+
+// components
+import Button from '../../presentationals/Button/Button';
+import Loader from '../../presentationals/Loader/Loader';
+import Input from '../../presentationals/Input/Input';
+
+//reducer
+import { loginUser } from '../../../store/modules/auth';
+
+//helper function
+import { validate } from '../../../utils/formValidator';
+
+export class Login extends Component {
+  state = {
+    userCredentials: { email: '', password: '' },
+    validationErrors: [],
+    invalidFields: [],
+  };
+
+  userInputHandler = event => {
+    const { validationErrors } = this.state;
+    const errors = validate(
+      { [event.target.name]: event.target.value },
+      validationErrors,
+    );
+
+    this.setState(
+      {
+        userCredentials: {
+          ...this.state.userCredentials,
+          [event.target.name]: event.target.value,
+        },
+        validationErrors: errors,
+      },
+      () => {
+        this.handleInvalidFields();
+      },
+    );
+  };
+
+  handleInvalidFields = () => {
+    const { validationErrors, userCredentials } = this.state;
+    const errors = [...validationErrors];
+    const fields = Object.keys(userCredentials);
+    const invalidFields = [];
+
+    errors.forEach(error => {
+      fields.forEach(field => {
+        if (error.includes(field) && !invalidFields.includes(field)) {
+          invalidFields.push(field);
+        }
+      });
+    });
+
+    this.setState({
+      invalidFields,
+    });
+  };
+
+  submitHandler = async event => {
+    event.preventDefault();
+
+    await this.validateForm();
```

Comment:
why do we have to await a function call here.

Reply:
Same as here